### PR TITLE
clang-tidy: sort includes LLVM style

### PR DESCRIPTION
**Pull request by neheb**
Found with llvm-include-order

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: samples/addmoddel.cpp [hunk 1]
```diff
@@ -2,11 +2,10 @@
 // addmoddel.cpp
 // Sample program showing how to add, modify and delete Exif metadata.
 
+#include <cassert>
 #include <exiv2/exiv2.hpp>
```
Comment (piponazo) [inline samples/addmoddel.cpp:6]: We want to keep `#include <exiv2/exiv2.hpp>` always on top. We would need to add an additional blank line to restrict the sorting

File: src/bigtiffimage.cpp [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-#include "error.hpp"
-#include "image_int.hpp"
-#include "enforce.hpp"
-
 #include <cassert>
-#include <limits>
 #include <iostream>
+#include <limits>
+
+#include "enforce.hpp"
```
Comment (piponazo) [inline src/bigtiffimage.cpp:7]: I also see here a problem. I think it is not written anywhere, but it was discussed in the past between the main contributors. We agreed on using the following style:

```
// Include exiv2 headers 
#include "enforce.hpp"
...

// Include 3rd party headers
...

// Include C++ and STL headers
...
```

I know that the situation is not perfect at the moment, but we do not want to break that sorting where we have it. 

In order to achieve that we would need to do some manual editing and introduce line breaks here and there.

File: src/crwimage_int.cpp [hunk 3]
```diff
@@ -1,13 +1,14 @@
 #include "crwimage_int.hpp"
+
+#include <cassert>
+#include <ctime>
```
Comment (piponazo) [inline src/crwimage_int.cpp:4]: Same issue here mixing the order of our own headers and the STD ones

File: src/exif.cpp [hunk 4]
```diff
@@ -25,21 +25,21 @@
  */
 // *****************************************************************************
 // included header files
-#include "config.h"
-
 #include "exif.hpp"
+
+#include <iostream>
```
Comment (piponazo) [inline src/exif.cpp:30]: same here

File: src/tiffcomposite_int.cpp [hunk 5]
```diff
@@ -24,22 +24,22 @@
  */
 // *****************************************************************************
 // included header files
-#include "config.h"
+#include "tiffcomposite_int.hpp"
```
Comment (piponazo) [inline src/tiffcomposite_int.cpp:27]: This one I like it. It seems like clang-tidy it is able to find the pair .h of the .cpp and place it on top :clap:

**Review by piponazo — APPROVED** [review]
Thanks again for your endless efforts! :clap: 

As I mentioned in one of the comments, we want to keep certain order in the headers. Are you willing to spend the extra time to satisfy that "non-written" rule? Otherwise, we can merge this as it is, and I will take care about that later.

**Comment by neheb** [comment]
I'll try to fix it up manually when I can.

**Comment by neheb** [comment]
Converted to draft to prevent anyone from merging as is.

**Comment by clanmills** [comment]
We should review this in v1.00 and either fix/merge or create a new PR using the tools used by @neheb

@neheb.  I'm really pleased to let you know that Exiv2 is scheduled for v1.00 on 2021-12-15.  We now have a team of about 8 to work on this.  We're aiming to assign issues at the end of April.  I would be delighted if you could bring your skills to the v1.00 project.  @piponazo is porting the unique_ptr code from branch old-master to 'main'.  I would like your clang_tidy and other improvements to be released in v1.00.

**Comment by neheb** [comment]
Hmm I've been out of the loop. What is this main branch based on?

**Comment by clanmills** [comment]
It's derived from 0.27-maintenance (v0.27.4) which we expect to ship 3 weeks early on 2021-04-30.

I'm working on the project plan at the moment and we hope to call a meeting on the weekend of 24 April or 1 May.  It looks the moment as though there will be four groups of 3 contributors in each group:

| Group | Members | Tasks |
|:--      |:--   |:--  |
| Code | Luis, Rosen, Pydera | C++11, jp2image.cpp, clang-xxxx tools. |
| Metadata | Milos, Christoph, Tom | BigTiff, Unified Metadata Container, Canon AFInfo, JPEG>64k |
| T&L |  Peter, Leonardo and AlexS. |  Test and localsation |
| PM&R | Alex E and Robin | User support, release engineering, project management |

The schedule looks like:

| Topic | Date | Purpose |
|:--   |:--   |:-- |
| Kickoff | 2021-05-01 | Zoom meeting to discuss the task list |
| Priority Complete | 2021-06-30 | Top priority features are complete |
| Week-end in Bavaria | 2021-09-18 | Have fun together |
| Code Complete        | 2021-09-30 | Finish all features to ship |
| RC1                          | 2021-10-21 | First cut |
| RC2/Freeze           | 2021-11-15 | No more changes |
| GM                      | 2021-12-15 | Success |

The detailed task list and priority is still being developed.  I hope each team will complete their **top priority** tasks by 2021-06-30.  If we have to drop lower priority tasks to make the schedule, we will.  If we hit **Priority Complete** in late June, success is certain.

The maximum commitment of time per contributor will be about 100 hours.  Most will be less.

**Pull request closed without merging**